### PR TITLE
[DE Team][Serverless] Case system action being added for rules

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Contributes to https://github.com/elastic/security-docs/issues/5457

Made the following changes to the [Create a detection rule](https://elastic-dot-co-docs-production-7m7ozw1rh-elastic-dev.vercel.app/current/serverless/security/rules-create) page:
- Revised step 5 at the start of the page so it reflects that users can set up actions for the rule
- Changed the title of the alert notifications section from "Set up alert notifications (optional)" to "Set up rule actions (optional)" and refreshed outdated images 
- Added a blurb about the case connector and linked to the connector docs
- Moved the response actions section one more level down so it's under the "Set up rule actions (optional)" section. This seemed appropriate, seeing that response actions are considered a sub-type of rule actions.

**Note:** This feature is being added in 8.16, not 8.15, so will be filing a separate ESS PR after 8.15.

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-ajo970vxt-elastic-dev.vercel.app

* Built with commit 8d8ece6ab31d9a4d5a555bf95ac072dc45f97dde

> Issues? Visit #next-docs in Slack

**Review by dhurley14 — DISMISSED** [review]
looks great thank you!!

**Comment by nastasha-solomon** [comment]
Reverting this back to draft because I just noticed that a couple of screenshots need to be refreshed:
- https://elastic-dot-co-docs-production-ajo970vxt-elastic-dev.vercel.app/L3ZlcmNlbC9wYXRoMC93b3JkbGFrZS1kb2Nz[…]detections-available-action-types.png
- https://elastic-dot-co-docs-production-ajo970vxt-elastic-dev.vercel.app/L3ZlcmNlbC9wYXRoMC93b3JkbGFrZS1kb2Nz[…]ctions-available-response-actions.png

Will re-open this PR for review once these images are updated.

**Review by joepeeples — APPROVED** [review]
LGTM!

File: docs/serverless/rules/rules-ui-create.mdx [hunk 1]
```diff
@@ -661,7 +661,7 @@ When configuring an ((esql)) rule's **<DocLink slug="/serverless/security/rules-
 
 <div id="rule-notifications"></div>
 
-## Set up alert notifications (optional)
+## Set up rule actions (optional)
 
 Use Actions to set up notifications sent via other systems when alerts are generated.
```
Comment (joepeeples) [inline docs/serverless/rules/rules-ui-create.mdx:666]: This probably doesn't need to be capitalized (fix in Asciidoc too?):
```suggestion
Use actions to set up notifications sent via other systems when alerts are generated.
```

File: docs/serverless/rules/rules-ui-create.mdx [hunk 2]
```diff
@@ -822,7 +824,7 @@ Example using the mustache "current element" notation `{{.}}` to output all the
 
 <div id="rule-response-action"></div>
 
-## Set up response actions (optional)
+### Set up response actions (optional)
 Use Response Actions to set up additional functionality that will run whenever a rule executes:
```
Comment (joepeeples) [inline docs/serverless/rules/rules-ui-create.mdx:828]: Similar, this probably doesn't need to be capitalized (fix in Asciidoc too?):
```suggestion
Use response actions to set up additional functionality that will run whenever a rule executes:
```

**Comment by dhurley14** [comment]
New screenshots:

rule_actions_response_actions_screenshot
rule_actions_docs_screenshot

**Review by joepeeples — APPROVED** [review]

**Comment by nastasha-solomon** [comment]
Changes are present in Serverless, so good to merge!

**Pull request merged**